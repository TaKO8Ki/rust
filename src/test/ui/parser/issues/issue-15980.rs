use std::io;

fn main(){
    let x: io::Result<()> = Ok(());
    match x {
        Err(ref e) if e.kind == io::EndOfFile {
            //~^ NOTE while parsing this struct
            //~| ERROR expected one of `!`, `.`, `::`, `=>`, `?`, or an operator, found `{`
            //~| NOTE expected one of `!`, `.`, `::`, `=>`, `?`, or an operator
            return
            //~^ ERROR expected identifier, found keyword `return`
            //~| NOTE expected identifier, found keyword
        }
        _ => {}
    }
}
