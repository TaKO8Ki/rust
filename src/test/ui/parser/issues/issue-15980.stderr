error: expected identifier, found keyword `return`
  --> $DIR/issue-15980.rs:10:13
   |
LL |         Err(ref e) if e.kind == io::EndOfFile {
   |                                 ------------- while parsing this struct
...
LL |             return
   |             ^^^^^^ expected identifier, found keyword
   |
help: escape `return` to use it as an identifier
   |
LL |             r#return
   |             ++

error: expected one of `!`, `.`, `::`, `=>`, `?`, or an operator, found `{`
  --> $DIR/issue-15980.rs:6:47
   |
LL |         Err(ref e) if e.kind == io::EndOfFile {
   |                                               ^ expected one of `!`, `.`, `::`, `=>`, `?`, or an operator
   |
help: try adding a fat arrow here
   |
LL |         Err(ref e) if e.kind == io::EndOfFile => {
   |                                               ++

error: aborting due to 2 previous errors

